python/ray/serve/_private/http_util.py

```diff
@@ -183,8 +187,8 @@ def get_messages_nowait(self) -> List[Message]:
         least one new message is available.
         """
         messages = []
-        while not self._message_queue.empty():
-            messages.append(self._message_queue.get_nowait())
+        while len(self._message_queue) > 0:
+            messages.append(self._message_queue.popleft())
```

Comment:
Is a `deque`'s `clear` method and a list's `extend` method threadsafe? If so, we could implement this as:

```suggestion
        messages.extend(self._message_queue)
        self._message_queue.clear()
```

Reply:
I think these are equivalent?

Reply:
I think you're right ([source](https://www.scaler.com/topics/extend-vs-append-python/)). For some reason, I though `extend` was faster than adding elements one-by-one. We can leave it as-is.